### ISSUE TITLE
Stop build -> Start build (OK) -> Stop build ->Start build -> "Build version x.x.x already exists. Update next build number or change version format."

### ISSUE THREAD

**Issue by bguiz** [appveyor/ci#376]
It should have just incremented the build number.

Otherwise, make the error display an option to right there - having to go to the setting page and change it each time is just painful.

**Comment by FeodorFitsner** [comment]
Are you getting this all the time or it was a one-time issue?

**Comment by bguiz** [comment]
It happens consistently

**Comment by FeodorFitsner** [comment]
OK, can you provide steps to reproduce that?

**Comment by bguiz** [comment]
1. (build is already running)
2. Stop build
3. Start build
4. (build starts successfully)
5. Stop build
6. Start build
7. "Build version x.x.x already exists. Update next build number or
change version format."
8. (build fails to start)

W: http://bguiz.com

**Comment by FeodorFitsner** [comment]
Is it a public project? Do you have custom version format or changing build version through API?

**Comment by bguiz** [comment]
> Is it a public project?

No, it is a private project.

> Do you have custom version format or changing build version through API?

No, it simply appears like this:

version: 4.1.{build}


This may be related to the other issue I opened -
https://github.com/appveyor/ci/issues/381 -
because, now that you mention it, I was almost certainly editing the
settings inbetween stopping and starting the builds.

W: http://bguiz.com

**Comment by FeodorFitsner** [comment]
Exactly, that issue is how you can reproduce. I'm closing this then.

**Comment by bguiz** [comment]
I still think that this is a bug though -
editing the config should not cause the version numbers to get mixed up,
especially since I don't edit the version number at all;
or even anything on the tab containing the version numbers, for that matter.


W: http://bguiz.com

**Comment by FeodorFitsner** [comment]
Well, it's kind of "optimistic concurrency" when working with database. You open the form edit it, the data behind the form is updated by someone else, then you save your form thus overwriting those changes with your own.

**Comment by bguiz** [comment]
Like I said in the other ticket,
if one page only edits a certain number of fields,
the rest of the fields should not be overwritten.
Ideally, each settings page should only update using the delta of what was
input in it,
instead of overwriting the whole lot.

In this particular use case, that makes the user experience particularly
painful,
because when setting up a project for the first time,
you really do want to be able to edit settings and
cancel+restart builds many times in quick succession until you get it right.


W: http://bguiz.com

**Comment by bguiz** [comment]
Put in a different way,
if each of these forms updates the database using stale values from other
forms,
it really does not matter what kind of concurrency model you have,
because stale value are just that.

Not sure why you have close this ticket - it has not been resolved.

W: http://bguiz.com

**Comment by FeodorFitsner** [comment]
OK, seems like I have some ideas how to prevent version override while editing the settings. Let's keep it open :)

**Comment by bguiz** [comment]
Thanks for looking into this!

W: http://bguiz.com

**Comment by saper** [comment]
Related: https://help.appveyor.com/discussions/problems/2989-a-push-of-40-tags-to-github-causes-job-numbering-disaster

**Comment by kveretennicov** [comment]
I encountered this issue today with https://ci.appveyor.com/project/kveretennicov/kallithea, where I was tweaking settings (mostly scripts) and then rebuilding same commit again and again. My build number pattern is just `{build}`.

At some point I started to suspect that the issue is due to multiple tabs being used to save setting updates. But it remained even after I left only 2 tabs with project pages - one for build settings and one for build console (and a few more tabs with AppVeyor documentation).
